### PR TITLE
Simplify `searchForHttpInText` method

### PR DESCRIPTION
**Pull request by mrcasals**
This PR simplifies code logic and makes it more comprehensible.

### CODE REVIEW

**Comment by txus** [comment]
Actually I'd say this PR implements the code for the first time, without needing a hand-written parser with manual backtracking and off the charts cyclomatic complexity. Kudos!

**Comment by jomo** [comment]
What the hell… :+1:

File: TMessagesProj/src/main/java/org/telegram/ui/ChatActivity.java [hunk 1]
```diff
@@ -2135,30 +2135,8 @@ public void onClick(DialogInterface dialogInterface, int i) {
     }
 
     private boolean searchForHttpInText(CharSequence string) {
```
Comment (jomo) [inline TMessagesProj/src/main/java/org/telegram/ui/ChatActivity.java]: You could also rename `string` to `sequence` or some other more accurate name, while you're on it.
`string.toString()` looks kinda weird…

**Comment by tjhorner** [comment]
Holy shit, that code was actually in prod? That's probably the worst I've ever seen.

Good job @mrcasals

**Comment by 3mph4515** [comment]
Maybe someone want to make some perfomance tests? :1234:

**Comment by DrKLO** [comment]
It was done like this to avoid memory allocation on every entered character, so what's the problem?
Anyway it's removed and replaced with regex and moved to background thread.

**Comment by jomo** [comment]
@DrKLO [that regex](https://github.com/DrKLO/Telegram/blob/74def22213846b1f90a26c86670035079dea032c/TMessagesProj/src/main/java/org/telegram/messenger/AndroidUtilities.java#L97-L123) doesn't match all URLs, for example [`http://to`](http://to). I don't expect "to" to be parsed as a link, but any valid URL prefixed with `http://` should match, no?

**Comment by jomo** [comment]
Additionally, it only seems to match `EXAMPLE.COM` for the string `HTTPS://EXAMPLE.COM`.
I guess that would end up being rendered as _HTTPS://[EXAMPLE.COM](http://example.com)_, effectively dropping the https protocol?

**Comment by DrKLO** [comment]
@jomo I'm using regex from android Linkify. I don't think that somebody will write http in upper case.
This regex used only for webpage preview request from server. For links highlighting we using server side text parser (cloud chats only).

**Pull request closed without merging**